### PR TITLE
Save tot as double instead of int

### PR DESCRIPTION
**Pull request by dvbruch**
Changed variable p_tot to double. 
In the Mupix chip analysis we use the signal and time of a hit to store two different timestamps. For one of them we need double precision, so we use p_tot as a double.

### CODE REVIEW

**Comment by tbisanz** [comment]
Just a minor side remark: You're using the EUTelGenericSparsePixel's signal and time value for two timestamps? I just wanted to point out that you could introduce a new pixel type for correctly storing this information. I agree that this might be a little more work, but if you're planning on using EUTelescope a lot in the future this is definitely the better solution.

**Comment by dvbruch** [comment]
@tbisanz At which level do you suggest to implement the pixel type? I was thinking of inheriting from the EUTelGenericSparsePixel so that all its functionalities are preserved for the noisy pixel finding, clustering, hit maker etc. But then to have the additional information of the time stamps if needed. Do you have any other suggestions?

**Comment by tbisanz** [comment]
From a simple object oriented approach I would agree - BUT we have the problem of serialisation of data to disk. This way we loose any run time type information and everything gets more difficult.

Lines like this: https://github.com/eutelescope/eutelescope/blob/master/src/EUTelProcessorNoisyPixelFinder.cc#L237 won't work, as you'd misinterpret the pixel type when retrieving it from the LCIO object. We additionally store the pixel type with this enum in the CellID field: https://github.com/eutelescope/eutelescope/blob/master/include/EUTELESCOPE.h#L608-L614 and we will have to use this to retrieve the correct pixel. Of course this should be some very fast lookup which goes into some utility class and/or namespace.

However, you probably still want to inherit from EUTelGenericSparsePixel - take a look at EUTelGeometricPixel. The interfacing between EUTelescope and LCIO is done via EUTelTrackerDataInterfacerImpl (look at how it is done for EUTelGenericSparsePixel  and EUTelGeometricPixel).

Basically, whenever you read the pixel from an LCIO collection you HAVE to use your pixel type. Once you have the object in memory, you can downcast w/o any problems.

**Comment by tbisanz** [comment]
I'll also spend some time thinking of an implementation for generic and fast retrieval vie the cellID field, but feel free to contribute. Also keep in mind, that all the values are stored as float in LCIO.

**Comment by dvbruch** [comment]
Thanks for your comments. 
So LCIO cannot store more than float precision and any doubles are converted? This is good to know.

**Comment by tbisanz** [comment]
Yes, exactly - although you can try to split up a double into two floats and put them together when reading from disk. Such functionality can be implemented in the EUTelTrackerDataInterfacerImpl, although you need to implement checks that ensure that proper endianness and such are taken care of.

**Comment by tbisanz** [comment]
The more I think about this the more I think we should finally move to C++14, writing this w/o smart ptrs seems like a very, very bad idea. I'll try to find some time to update the EUTelTrackerDataInterfacerImpl to use smart ptrs and refactor some other code as well. Please use them in your new changes and try to update old code. Don't forget to call release() when handing over ownership to LCIO!

**Comment by tbisanz** [comment]
Sorry for spamming that much - I just want to keep all the ideas/plans at one place and let you know at the same time. I'll update https://github.com/eutelescope/eutelescope/blob/master/include/EUTelTrackerDataInterfacer.h#L54 to actually retrieve the Pixel itself not a ptr.  All those heap-allocations are utterly useless.

**Comment by dvbruch** [comment]
Ok, I have an implementation of a MuPixel type working now with the current DataInterfacer on this branch: https://github.com/dvbruch/eutelescope/tree/NewPixelTypeWithTimestamps
I'll work on passing the double through LCIO now and wait until you changed things to adjust with the new DataInterfacer. 
Do you mean the call to release() in the converter plugin in eudaq when creating the raw data collection? We do that for the Mupix.

**Comment by tbisanz** [comment]
I mean you must not forgive to pass ownership of the LCIO collection pointers to LCIO when using smart pointers and it is required.

I'll close this PR, once your new branch is ready to merge please open a new PR.
